python/thunder/imgprocessing/register.py

```diff
@@ -1,252 +1,316 @@
+    and passing any additional keyword arguments. The algorithm
+    can then be used to fit registration parameters on an Images object.
 
-        checkparams(method, ["crosscorr"])
+    Parameters
+    ----------
+    method : string
+        A registration method, options include 'crosscorr' and 'planarcrosscorr'
+    """
 
-        if method == "crosscorr":
-            return super(Register, cls).__new__(CrossCorr)
-        else:
-            raise Exception('Registration method not recognized')
+    def __new__(cls, method, **kwargs):
 
-    def get_transform(self, im, ref):
-        raise NotImplementedError
+        from thunder.imgprocessing.regmethods.crosscorr import CrossCorr, PlanarCrossCorr
 
-    def apply_transform(self, im, transform):
-        raise NotImplementedError
+        REGMETHODS = {
+            'crosscorr': CrossCorr,
+            'planarcrosscorr': PlanarCrossCorr
+        }
 
-    def setFilter(self, filter='median', param=2):
-        """
-        Set a filter to apply to images before registration.
+        checkparams(method, REGMETHODS.keys())
 
-        The filtering will be applied to both the reference and
-        image to compute the transformation parameters, but the filtering
-        will not be applied to the images themselves.
+        return REGMETHODS[method](kwargs)
 
-        Parameters
-        ----------
-
-        filter : str, optional, default = 'median'
-            Which filter to use (options are 'median' and 'gaussian')
-
-        param : int, optional, default = 2
-            Parameter to provide to filtering function (e.g. size for median filter)
+    @staticmethod
+    def load(file):
+        """
+        Load a registration model from a file specified using JSON.
 
         See also
         --------
-        Images.medianFilter : apply median filter to images
-        Images.gaussianFilter : apply gaussian filter to images
-
+        RegisterModel.save : specification for saving registration models
+        RegisterModel.load : specifications for loading registration models
         """
 
-        checkparams(filter, ['median', 'gaussian'])
+        return RegisterModel.load(file)
 
-        if filter == 'median':
-            from scipy.ndimage.filters import median_filter
-            self._filter = lambda x: median_filter(x, param)
 
-        if filter == 'gaussian':
-            from scipy.ndimage.filters import gaussian_filter
-            self._filter = lambda x: gaussian_filter(x, param)
+class RegisterMethod(object):
+    """
+    Base class for registration methods
+    """
 
-        return self
+    def __init__(self, *args, **kwargs):
+        pass
 
-    def filter(self, im):
-        """
-        Apply filtering, and if not set, return image unchanged
-        """
-
-        if hasattr(self, '_filter'):
-            return self._filter(im)
-        else:
-            return im
+    def getTransform(self, im, ref):
```

Comment:
Moving `ref` to an argument of `__init__` may make more sense as some registration methods may not use references.